htdocs/core/ajax/selectobject.php

```diff
@@ -53,7 +53,7 @@
 	httponly_accessforbidden('Bad value for param htmlname');
 }
 
-if (!empty($objectfield)) {
+if (!empty($objectfield) && empty($objectdesc)) {
```

Comment:
I don't understand why you need this. Allowing a user to force himself the value of objectdesc as parater is a security hole.
When we call ajax/selectobject, the main parameter is objectfield. And from it, we should be able to retreive all information of objectdesc.
The objectdesc should not be provided as parameter (a end user must not be able to bypass definition nown by dolibarr).
What value do you have for $objectfield ?
And what value do you have in the definition table llx_extrafields of the extrafield mentionned into $objectfield ?